torch/optim/sparse_adam.py

```diff
@@ -97,11 +97,36 @@ def step(self, closure=None):
 
         return loss
 
-SparseAdam.__doc__ = r"""Implements lazy version of Adam algorithm suitable for sparse tensors.
-
-    In this variant, only moments that show up in the gradient get updated, and
-    only those portions of the gradient get applied to the parameters.
-
+SparseAdam.__doc__ = r"""SparseAdam implements a lazy version of the Adam algorithm
+    suitable for sparse gradients. Please note that SparseAdam is NOT equivalent to
+    Adam "but for sparse gradients", and we understand this may be confusing.
+
+    This is a semantically different optimizer where only moments that show up in the
+    gradient get updated, and only those portions of the gradient get applied to the
+    parameters. This is an optimization, since the 0-values in the gradient will be
+    skipped, and one can presume that this is only an optimization in semantically
+    sparse tensors, meaning there are lots of 0-values that are skipped. We implement
+    the optimization by using the sparse layout to do the update computations. This
+    does NOT mean that SparseAdam is especially meant for tensors with a sparse layout,
+    as some of our other optimizers can handle tensors with a sparse layout as well.
+
+    SparseAdam is intended for a narrow subset of parameters, specifically parameters
+    of a dense layout with gradients of a sparse layout. This is a special case where
+    the module backwards produces grads with a sparse layout and we can thus take
+    advantage of the sparsity of the grads. One example NN module that behaves as such
+    is `nn.Embedding(sparse=True)`.
+
+    If you suspect your gradients are semantically sparse (but do not have sparse layout),
```

Comment:
it would be useful for users to know that this potentially might be implemented in future releases